### ISSUE TITLE
Windows release not working?

### ISSUE THREAD

**Issue by dragons4life** [xylosper/bomi#266]
http://i.imgur.com/5oSp5aK.png
http://i.imgur.com/0Z7RFD5.png
Video does not play, only sound comes out. most of the time the window just stays black.
<img src='http://i.imgur.com/WjvlkkE.png'/>
Also when closing the player the window goes white and does't close or hides to tray.
<img src='http://i.imgur.com/vtfWswr.png'/>

**Comment by xylosper** [comment]
Video corruption is mainly caused by  bad/old graphics driver support.
What is your graphics card? Also, please check if you have latest driver.

**Comment by dragons4life** [comment]
Intel HD4000, It's from a Laptop.
Also i doubt it because I've used bakamplayer and the normal mpv version, without any problems.

**Comment by xylosper** [comment]
Oh, you can test mpv, that's good. Can you try `mpv --vo=opengl` or `mpv --vo=opengl-hq`?

**Comment by dragons4life** [comment]
Sorry, I'll be honest.
I did't understand what you mean with opengl, but yes I have use mpv before.

or do you want me to test something with it?

**Comment by xylosper** [comment]
You have to test mpv with --vo=opengl option if you compare bomi with mpv.
Do you know how to give option to mpv?

**Comment by dragons4life** [comment]
To the normal version? well it never really gives me options reason or rather don't know how to use them/enable them.

**Comment by xylosper** [comment]
Thank you. In that case, can you provide a log file?
Please set 'Text File' to 'debug' in Preferences > General > Application > Log Item.
Also, check the path for file and restart bomi and play whatever the debug log will be generated at the give path.
Then please post the content of the log file.

**Comment by dragons4life** [comment]
There is no "preferences" option
<img src='http://i.imgur.com/b6gnyMe.png'/>
I am right clicking the player for this to come up, this same menu comes up when I click the tray icon as well.
 the only thing close to a Log file is "log viewer" but it's just this:
<img src='http://i.imgur.com/xZZihDv.png'/>

**Comment by xylosper** [comment]
'Preferences' is under 'Tools' menu.

**Comment by dragons4life** [comment]
@xylosper You are right, I apparently did't see it.
Here is the log file: https://a.pomf.se/ldpzfg

<s>I enabled it > quit player > check if it was enabled again > played video (at least for 20 seconds</s>
Just ask if you want me to leave the video file to play for a longer time.

**Comment by xylosper** [comment]
Your link is broken.

**Comment by dragons4life** [comment]
http://a.pomf.se/ldpzfg How about now? pomf has some weird issues with "https"
http://www.filedropper.com/ldpzfg here is another link just in case.

**Comment by xylosper** [comment]
It seems a bug of intel's driver (https://communities.intel.com/thread/48164).
I'm trying to find a workaround.

**Comment by dragons4life** [comment]
On my desktop, the video actually plays.
But there are some bugs.
-bringing up preference menu freezes/lagg video/program
-opening a video makes the program lagg/freeze
-Player still doesn't close when closing windows nor does it go to tray, it just stays with a white screen
http://a.pomf.se/brlpzz.png
-Lots of laggs issue when opening (right-click) menu and browsing trough it's features
-Black bars http://a.pomf.se/ekwziz.png on fullscreen? (fixed by windows > 100% or Proper Window)
-Black bars if not fullscreen http://a.pomf.se/apuave.png (on the side)

There is a lot of buggs regarding lagg and black bars, I also did't found no feature to make a video window take 75% or 50% or just use a custom size (like in PotPlayer http://a.pomf.se/qaufcj.png) 

Main problems:
-Window is too small and there is no good way to easyly manipulate the size (like in potplayer/mpc-hc), as of right now this is how the windows size of the program when it opens up http://a.pomf.se/clpfjn.png (small and with blackbars) Only way to fix this and give it a decent size is using "Windows > Proper size" and even than like i said earlier you still see black bars http://a.pomf.se/abfker.png (Honestly would prefer if you guys added a feature like MPC-HC and PotPlayer for the windows settings [if you weren't planning too])

TL;DR: lots of laggs/freezes and black bar.
sorry if sometimes I don't express the issues correctly, english is not my main language.

**Comment by xylosper** [comment]
Intel driver issue has been fixed.
I cannot open pomf.se urls. I'm not sure but I'm geussing that it's banned by my internet provider or korean goverment which is totally ununderstandable.
I think the issue you said 'black bar' is what's already fixed in latest source. Please test with next version.
Also, I'm planning to make the menus of 'proper size/100%/200%/300%/400%' customizable for next version.

**Comment by dragons4life** [comment]
Have you tried taking off "https://" from the url(s)? 
If they still don't work use: http://imgur.com/a/cEziL (rearranged them to match the same order)

Also it's more important to make a window 50%/75% size of the original video than a customizable one (IMO).

would it be hard to add a "sharpen" feature? I believe vanilla mpv has one:
http://mpv.io/manual/master/#video-filters-unsharp[ 

Of course, whenever you have the time too and if you want to.

**Comment by xylosper** [comment]
> Have you tried taking off "https://" from the url(s)? 

Your last links don't contain https, so it's not about ssl.

> Also it's more important to make a window 50%/75% size of the original video than a customizable one (IMO).

I don't think so. It makes much more sense to calculate proper size with monitor size not video (and that's what 'Proper Size' does).

Do you really want to see a video in exactly 50% of its original size?
Or do you just want to see it in a specific window size?
I think, the latter is your answer. If someone wants to display the window in smaller window size than original video size, that's because he or she wants to see it a certain small size and 'THE certain small size' would not be defined by video size. Noone would calculate window size from video output size.

That's why bomi provides a 'Proper Size' menu.
With current implemetation, proper size is fixed as 15% area of display.
I think what bomi really lacks is an option to define 'proper size'.

**Comment by xylosper** [comment]
> would it be hard to add a "sharpen" feature? 

Have you tried Unsharp Masking in 'Video > Interpolator'?

**Comment by dragons4life** [comment]
I have not, I will check it out in a bit.

**Comment by xylosper** [comment]
New version binary for Windows uploaded. Could you test it?

**Comment by dragons4life** [comment]
Didn't find any bugs for this release on my laptop, as of now.
Except this one, it happens when you quit the window, it just goes all black </img src='http://i.imgur.com/lyWutec.png'/>
(easily fixed by going to preferences and un-checking "hide the window rather than close")

Going to test on Desktop in a bit, if you want me to test anything for you, just ask me.

**Comment by dragons4life** [comment]
Desktop 964-bit):
-Video/windows lags when you move the window around (this might be a thing most people don't do, but personally I do it a lot, since I use livestreamer sometimes)
http://a.pomf.se/vbwmvo.mp4 (don't know if you can actually tell trough a video)
-The white screen when closing window still happens on my desktop (however "the fix" works on here too)
http://a.pomf.se/vcwxju.mp4
-Quality?
<img src="http://a.pomf.se/fxmnvr.png"/> (MadVR + PotPlayer)
<img src="http://a.pomf.se/bajvvh.png"/> (bakamplayer)
<img src="http://a.pomf.se/kftpht.png"/> (bomi)
Just in case you think this only happens when taking a screenshot:
http://a.pomf.se/itnusi.png

<s><i>this happened to me on my laptop as well, but I just assumed it was just my laptop being bad.</i></s>

**Comment by xylosper** [comment]
'white screen on closed' fixed.

**Comment by xylosper** [comment]
> -Quality? (debanding/dithering?)

This was a valid issue(#282) and fixed.

**Comment by dragons4life** [comment]
I'll be honest the windows lag (when moving the window around, only happens when video is playing) doesn't happen when I use my integrated Intel GPU and does not happen on my laptop.

I know for a fact it's my GPU, but this doesn't happen with the other 5 players I tried, I also found out that said thing stop happening once I lower my entire Nvidia GPU settings to performance instead of performance.

<img src="http://a.pomf.se/wosghk.png"/>

So maybe the problem isn't at my end but the program itself?

**Comment by xylosper** [comment]
If it gets slow even if you move window by dragging the window title bar, it is the hardware problem. If it gets slow down only if you move window by dragging video screen, it is the problem of software. I can tell because I can't diwnload pomf.se urls

**Comment by dragons4life** [comment]
Here is another link, I keep thinking you are seeing these links.
http://www.4shared.com/video/O74R0zDlba/vbwmvo.html
It's kind of blurry since I couldn't find anywhere to host it.

**Comment by xylosper** [comment]
I cannot notice any difference in your video... btw, I figure out that a.pomf.set doesn't work for me but I can download b.pomf.se.

Anyway please follow next steps and then test again if anything changed.

1. Press Win+R and run `control.exe %windir%\system32\sysdm.cpl` to open system properties.
2. Select 'Advanced' tab, then click the 'Environment Variables' button near the bottom of that tab.
3. In 'User variables for ...' group, click 'New' button and input
  * Variable name: QSG_RENDER_LOOP
  * Variable value: threaded
4. Click 'Ok' and test bomi

**Comment by xylosper** [comment]
I think  it's fixed. Please test it when next version released.

**Comment by dragons4life** [comment]
Sorry that did not work, I even restarted my computer.

However don't worry too much about it and try to add more stuff people recommend or you were planning to, it's probably only me having this problem.

**Comment by xylosper** [comment]
New version released. Please test it.

**Comment by dragons4life** [comment]
Bomi 0.9.6
_________

-When dragging the program trough the window tittle instead of the video (center) it does not snap to edges, also sometimes resets itself to the center position

-<img src='http://i.imgur.com/yY5QVbI.png'/>

**Comment by xylosper** [comment]
> -When dragging the program trough the window tittle instead of the video (center) it does not snap to edges, also sometimes resets itself to the center position

Fixed.

> -Drop down menus are out of place? (this also happens with the video filters, and probably anything with a drop down menu)

Can't reproduce.
Do you happen to use some kind of tweak for window style or window frame?

> -%RAND_n% is not working correctly , unless this is the way it's supposed to name the screenshots.

n is an positive integer which indiates digits.
For instance, RAND_5 will produce a random number in 5 digits.

> -Weird bug were sometimes livestreamer streams don't open up

I'll check it later.

> EDIT: The lagg when dragging the player/window still happens if I am playing a 60fps video at both 1080p and 720p resolution. (Entering fullscreen mode and going back to windowed mode sometimes fixes the bug).

When you saying 'lagg', does it mean stuttering of video playback, or not-smooth-moving of window?

**Comment by dragons4life** [comment]
@xylosper 
Apparently the drop down menu bug fixed by itself on both my laptop and desktop
___
I tough %RANd_n% was made so that the entire screenshot became random numbers, oh well.
___
"lagg" as in fps lagg in the video player itself (the window not the actual video).
I wish i could capture it but everytime I do, the entire lagg just doesn't show, don't know why, i am even capturing at @60fps.

However it's not as annoying because sometimes it randomly disappears.

**Comment by xylosper** [comment]
I've never used livestreamer. How do you open livestreamer with bomi?

> I tough %RANd_n% was made so that the entire screenshot became random numbers, oh well.

Just put %RAND_n% with some number for n.

> Also do you mind making it so that whenever you have the sound muted and you increase the volume it automatically unmute itself?

An otpion to do so added.

> Is this supposed to be normal?
https://sendvid.com/x9mb0mbp

If you are talking about wrong position of combo box list, of course it's not normal.

**Comment by dragons4life** [comment]
<blockquote>If you are talking about wrong position of combo box list, of course it's not normal.</blockquote>
No, not only that but look at the slides, in unsharpen you can only use 3 or 5? no 4? and in some others it just skips to the last and first ending numbers if you click.

<blockquote>I've never used livestreamer. How do you open livestreamer with bomi?</blockquote>
You can use CMD or use a GUI

https://github.com/bastimeyer/livestreamer-twitch-gui/releases
Would suggest to just use this, go to settings, select video player, and done.

https://www.youtube.com/watch?v=o473rRwoJw0
Without GUI only the livestreamer program.

**Comment by xylosper** [comment]
> No, not only that but look at the slides, in unsharpen you can only use 3 or 5? no 4? and in some others it just skips to the last and first ending numbers if you click.

Yes, they're normal.
Unsharp masking taks 3 or 5 for vaild radius.
About 'skip', it's just following the standard behavior when a slider is clicked. Because there's only a few ticks, it may look 'skipping'. However, since I think this could be inconvenient, made a change to enforce page step to 1 for radius.

> You can use CMD or use a GUI

I need to know exactly what you do in order to test your problem.

**Comment by dragons4life** [comment]
<blockquote>I need to know exactly what you do in order to test your problem.</blockquote>
I use livestreamer GUI

Settings > Videoplayer settings > videoplayer > add bomi > look for a stream

**Comment by xylosper** [comment]
@dragons4life 
Have you tested livstreamer with latest version(0.9.10) with youtube-dl?
Or, if you don't mind this, please close this issue.
